src/libstd/io/buffered.rs

```diff
@@ -323,14 +326,18 @@ impl<R: Seek> Seek for BufReader<R> {
             } else {
                 // seek backwards by our remainder, and then by the offset
                 self.inner.seek(SeekFrom::Current(-remainder))?;
-                self.pos = self.cap; // empty the buffer
+                // Empty the buffer
+                self.cap = 0;
+                self.pos = 0;
```

Comment:
Why do we need to mark the buffer as empty here, when we do the same further down?

Reply:
Never-mind just saw the conversation above!